index.html

```diff
@@ -0,0 +1,571 @@
+
+Determine the name of the core specification within the document *or* fail the *Has Core Feature* validation.
+
+It is possible to [rename the core feature](#renaming-core-itself) within a document. This process determines the actual name for the core feature if one is present.
+
+**Formal Specification**
+- For each directive `d` on the SchemaDefinition within the Document,
+  - If `d` has a `feature:` argument whose value is "https://lib.apollo.dev/core/v0.1", *and either*:
+    - &#8230;`d` has an `as:` argument whose value is equal to `d`'s name
+    - &#8230;*or* `d` does not have an `as:` argument and `d`'s name is `core`
+    - *then* **Return** `coreName` = `d`'s name
+- If no matching directive was found, the ***Has Core Feature* validation fails**.
+
+### Namespace Collection *validates* Name Uniqueness
+
+Collect a map of `featureName: String -> Directive`, where `Directive` is a `@core(feature:)` Directive which introduces some feature named `featureName` into the document.
+
+**Formal Specification**
+- Given [`coreName`](#bootstrapping-validates-has-core-feature)
+- Let `namespaces` be a map of `featureName: String -> Directive`, initially empty.
+- For each directive `d` named `coreName` on the SchemaDefinition within the Document,
+  - Let `name` be the spec's [name](#prefixing) as specified by the directive's `as:` argument or, if the argument is not present, the default name specified in the [feature url](#feature-urls).
+  - If `name` exists within `namespaces`, the ***Name Uniqueness* validation fails**.
+  - Insert `name => d` into `namespaces`
+- **Return** `namespaces`
+
+**Explanatory Text**
+
+Prefixes, whether implicit or explicit, must be unique within a document. Valid:
+
+<view-of src=prefixing.graphql#schema[0]
+  title="Example of unique prefixes"
+></view-of>
+
+It is also valid to reference multiple versions of the same spec under different prefixes:
+
+<view-of src=prefix-uniqueness.graphql#schema[0]
+  title="Explicit prefixes allow multiple versions of the same spec to coexist within a Document"
+></view-of>
+
+Without the explicit `as:`, the above would be invalid:
+
+<view-of src=prefix-uniqueness.graphql#schema[1]
+  title="Counter-example of non-unique prefixes with multiple versions of the same spec"
+  class=counter-example
+></view-of>
+
+Different specs with the same prefix are also invalid:
+
+<view-of src=prefix-uniqueness.graphql#schema[2]
+  title="Counter-example of non-unique prefixes"
+  class=counter-example
+></view-of>
+
+### Assign Namespaces
+
+Create a map of `element: Any Named Element -> featureName: String | null`, associating every named schema element within the document with the name of a feature, or `null` if it is not associated with a feature.
+
+**Formal Specification**
+- Given [`namespaces`](#namespace-collection-validates-name-uniqueness)
+- Let `assignments` be a map of `element: Any Named Element -> featureName: String | null`, initally empty
```

Comment:
I don't see the value in structuring this as "create a map from all elements to the answer" rather than just "a function to calculate an answer from an element, which may be memoized". (You can define functions in your spec.)

I also think `featureNames` would be a better name than `assignments`.